src/util.rs

```diff
@@ -76,6 +111,26 @@ impl std::ops::SubAssign for Timeout {
     }
 }
 
+impl From<Timeout> for Deadline {
+    fn from(t: Timeout) -> Deadline {
+        if let Timeout::After(dur) = t {
+            Deadline::new(Some(cmp::min(Deadline::MAX_FLUSH_DURATION, dur)))
```

Comment:
this shouldn't be needed anymore as we've `remaining_millis_i32()` now.

Reply:
Fixed in ab7b4af21590408e4ba3562994312e9889a1b934.